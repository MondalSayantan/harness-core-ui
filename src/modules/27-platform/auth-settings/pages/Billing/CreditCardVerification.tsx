/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useEffect, useState } from 'react'
import { Layout } from '@harness/uicore'
import type { SubscribeViews, SubscriptionProps } from '@common/constants/SubscriptionTypes'
import { CreditCard, Category } from '@common/constants/TrackingConstants'
import { useTelemetry } from '@common/hooks/useTelemetry'
import type { InvoiceDetailDTO } from 'services/cd-ng'
import PaymentMethod from '@auth-settings/components/Subscription/PaymentMethod/PaymentMethod'
import { FooterCreditCard } from './FooterCreditCard'

interface CreditCardVerificationProp {
  subscriptionProps: SubscriptionProps
  setView: (view: SubscribeViews) => void
  setInvoiceData: (value: InvoiceDetailDTO) => void
  setSubscriptionProps: (props: SubscriptionProps) => void
  onClose: () => void
  clientSecret: string
}
export default function CreditCardVerification({
  subscriptionProps,
  setSubscriptionProps,
  onClose,
  clientSecret
}: CreditCardVerificationProp): JSX.Element {
  const { trackEvent } = useTelemetry()
  const [savechecked, setSaveChecked] = useState<boolean>(false)
  useEffect(() => {
    trackEvent(CreditCard.CalculatorStripeElementLoaded, {
      category: Category.CREDIT_CARD,
      module
    })
    return () => {
      trackEvent(CreditCard.CalculatorPaymentMethodStepExited, {
        category: Category.CREDIT_CARD,
        module
      })
    }
  }, [])

  return (
    <Layout.Vertical>
      <PaymentMethod
        setSaveChecked={(value: boolean) => {
          setSaveChecked(value)
        }}
        nameOnCard={subscriptionProps.paymentMethodInfo.nameOnCard}
        setNameOnCard={(value: string) => {
          setSubscriptionProps({
            ...subscriptionProps,
            paymentMethodInfo: {
              ...subscriptionProps.paymentMethodInfo,
              nameOnCard: value
            }
          })
        }}
        setValidCard={(value: boolean) => {
          setSubscriptionProps({
            ...subscriptionProps,
            isValid: value
          })
        }}
      />
      <FooterCreditCard
        onClose={onClose}
        isValid={subscriptionProps.isValid && savechecked}
        nameOnCard={subscriptionProps.paymentMethodInfo.nameOnCard}
        subscriptionId={subscriptionProps.subscriptionId}
        clientSecret={clientSecret}
      />
    </Layout.Vertical>
  )
}
