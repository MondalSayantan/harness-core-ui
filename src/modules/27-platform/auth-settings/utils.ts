/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { get } from 'lodash-es'
import type { CreateIpAllowlistConfigProps, UpdateIpAllowlistConfigProps } from '@harnessio/react-ng-manager-client'

import type { IIPAllowlistForm } from '@auth-settings/interfaces/IPAllowlistInterface'
import { SourceType } from '@auth-settings/interfaces/IPAllowlistInterface'
import { CurrencyType } from '@common/constants/SubscriptionTypes'

export function getAmountInCurrency(currency: CurrencyType, amount: number): string {
  switch (currency) {
    case CurrencyType.USD: {
      return amount.toLocaleString('en-US', {
        style: 'currency',
        currency: 'USD'
      })
    }
  }

  return ''
}

// dividing the actual amount by 100 as amount numbers are in cents
export function getDollarAmount(amount?: number, isYearly?: boolean): number {
  if (isYearly) {
    return (amount || 0) / 1200
  }
  return (amount || 0) / 100
}

export const getErrorMessage = (error: any): string =>
  get(error, 'data.error', get(error, 'data.message', error?.message))

export const buildCreateIPAllowlistPayload = (data: IIPAllowlistForm): CreateIpAllowlistConfigProps => {
  const allowSourceType = []
  if (data.allowSourceTypeUI) {
    allowSourceType.push(SourceType.UI)
  }
  if (data.allowSourceTypeAPI) {
    allowSourceType.push(SourceType.API)
  }
  return {
    body: {
      ip_allowlist_config: {
        allowed_source_type: allowSourceType,
        description: data?.description,
        enabled: false, // always false while creating
        identifier: data.identifier,
        ip_address: data.ipAddress,
        name: data.name,
        tags: data?.tags
      }
    }
  }
}

export const buildUpdateIPAllowlistPayload = (
  data: IIPAllowlistForm,
  overrideFields?: any
): UpdateIpAllowlistConfigProps => {
  const allowSourceType = []
  if (data.allowSourceTypeUI) {
    allowSourceType.push(SourceType.UI)
  }
  if (data.allowSourceTypeAPI) {
    allowSourceType.push(SourceType.API)
  }
  return {
    'ip-config-identifier': data.identifier,
    body: {
      ip_allowlist_config: {
        allowed_source_type: allowSourceType,
        description: data?.description,
        enabled: data?.enabled,
        identifier: data.identifier,
        ip_address: data.ipAddress,
        name: data.name,
        tags: data?.tags,
        ...overrideFields
      }
    }
  }
}

/**
 * Converts minutes to higher dimensions of time such as months, days and hours
 *
 * Examples:
 * formatMinutesToHigherDimensions(150);    // Output: "2 hours 30 minutes"
 * formatMinutesToHigherDimensions(4320);   // Output: "3 days"
 * formatMinutesToHigherDimensions(45000);  // Output: "1 month 1 day 6 hours"
 * @param minutes
 * @returns a string in higher dimensions
 */
export const formatMinutesToHigherDimensions = (minutes: number | undefined): string => {
  if (minutes === undefined || minutes < 0) {
    return 'Invalid input'
  }

  const MINUTES_IN_AN_HOUR = 60
  const HOURS_IN_A_DAY = 24
  const DAYS_IN_A_MONTH = 30

  const result = []

  if (minutes >= MINUTES_IN_AN_HOUR * HOURS_IN_A_DAY * DAYS_IN_A_MONTH) {
    const months = Math.floor(minutes / (MINUTES_IN_AN_HOUR * HOURS_IN_A_DAY * DAYS_IN_A_MONTH))
    result.push(`${months} month${months > 1 ? 's' : ''}`)
    minutes %= MINUTES_IN_AN_HOUR * HOURS_IN_A_DAY * DAYS_IN_A_MONTH
  }

  if (minutes >= MINUTES_IN_AN_HOUR * HOURS_IN_A_DAY) {
    const days = Math.floor(minutes / (MINUTES_IN_AN_HOUR * HOURS_IN_A_DAY))
    result.push(`${days} day${days > 1 ? 's' : ''}`)
    minutes %= MINUTES_IN_AN_HOUR * HOURS_IN_A_DAY
  }

  if (minutes >= MINUTES_IN_AN_HOUR) {
    const hours = Math.floor(minutes / MINUTES_IN_AN_HOUR)
    result.push(`${hours} hour${hours > 1 ? 's' : ''}`)
    minutes %= MINUTES_IN_AN_HOUR
  }

  if (minutes > 0) {
    result.push(`${minutes} minute${minutes > 1 ? 's' : ''}`)
  }

  return result.join(' ')
}
