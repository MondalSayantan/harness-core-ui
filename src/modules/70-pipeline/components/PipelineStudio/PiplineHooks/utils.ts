/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { defaultTo, flatMapDeep, identity, map, sortBy, sortedUniq } from 'lodash-es'
import { VariableMergeServiceResponse, VariableResponseMapValue } from 'services/pipeline-ng'

export interface Params {
  metadataMap: Record<string, VariableResponseMapValue>
  localStageKeys: string[]
}

export interface AllExpressionsObj {
  expressionsList: string[]
  outputExpressions: string[]
  extraExpressions: string[]
  extraOutputExpressions: string[]
}

/**
 * Traverse over stage and find out all local fqn
 */
export function traverseStageObject(
  jsonObj: Record<string, any>,
  metadataMap: Required<VariableMergeServiceResponse>['metadataMap']
): string[] {
  const keys: string[] = []
  if (jsonObj !== null && typeof jsonObj === 'object') {
    Object.entries(jsonObj).forEach(([_key, value]) => {
      keys.push(...traverseStageObject(value, metadataMap))
    })
  } else if (metadataMap[jsonObj]) {
    keys.push(jsonObj)
  }
  return keys
}

export function pickExpressionsFromMetadataMap(
  metadataMap: Record<string, VariableResponseMapValue>,
  localStageKeys: string[],
  key: 'yamlProperties' | 'yamlOutputProperties'
): string[] {
  return sortedUniq(
    sortBy(
      map(metadataMap, (item, index) =>
        index && localStageKeys.indexOf(index) > -1
          ? defaultTo(item[key]?.localName, '')
          : defaultTo(item[key]?.fqn, '')
      ).filter(p => p),
      identity
    )
  )
}

export function pickExtraExpressionsFromMetadataMap(
  metadataMap: Record<string, VariableResponseMapValue>,
  localStageKeys: string[],
  key: 'properties' | 'outputproperties'
): string[] {
  return sortedUniq(
    sortBy(
      flatMapDeep(metadataMap, (item, index) => {
        const properties = defaultTo(item.yamlExtraProperties?.[key], [])

        return properties.map((p: Record<string, string>) =>
          index && localStageKeys.indexOf(index) > -1 ? p.localName : p.fqn
        )
      }).filter(p => p),
      identity
    )
  )
}

export const getAllExpressionsFromMetadataMap = ({ metadataMap, localStageKeys }: Params): AllExpressionsObj => ({
  expressionsList: pickExpressionsFromMetadataMap(metadataMap, localStageKeys, 'yamlProperties'),
  outputExpressions: pickExpressionsFromMetadataMap(metadataMap, localStageKeys, 'yamlOutputProperties'),
  extraExpressions: pickExtraExpressionsFromMetadataMap(metadataMap, localStageKeys, 'properties'),
  extraOutputExpressions: pickExtraExpressionsFromMetadataMap(metadataMap, localStageKeys, 'outputproperties')
})
