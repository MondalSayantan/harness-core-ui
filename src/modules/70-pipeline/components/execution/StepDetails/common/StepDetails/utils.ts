/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { ResponseApprovalInstanceResponse } from 'services/pipeline-ng'

export function encodeURIWithReservedChars(uri: string): string {
  return encodeURIComponent(uri).replace(/[;,/?:@&=+$#]/g, function (c) {
    return '%' + c.charCodeAt(0).toString(16)
  })
}

export interface ExecutionMetadataType {
  approvalInstanceId: string
  mock?: {
    data?: ResponseApprovalInstanceResponse
    loading?: boolean
  }
}
