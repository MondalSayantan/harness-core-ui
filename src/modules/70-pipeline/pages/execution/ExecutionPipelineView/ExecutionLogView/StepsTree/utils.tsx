import { defaultTo } from 'lodash-es'
import { ExecutionPipelineNode } from '@pipeline/components/ExecutionStageDiagram/ExecutionPipelineModel'
import { ExecutionNode, InterruptEffectDTO } from 'services/pipeline-ng'

export function getRetryInterrupts(step: ExecutionPipelineNode<ExecutionNode>): InterruptEffectDTO[] {
  return defaultTo(step?.item?.data?.interruptHistories, defaultTo(step?.group?.data?.interruptHistories, [])).filter(
    row => row.interruptType === 'RETRY'
  )
}
