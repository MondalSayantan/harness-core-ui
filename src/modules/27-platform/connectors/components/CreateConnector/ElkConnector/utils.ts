/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormData } from '@platform/connectors/interfaces/ConnectorInterface'
import { AuthTypes } from '@platform/connectors/pages/connectors/utils/ConnectorHelper'
import { UseStringsReturn } from 'framework/strings'
import type { ConnectorConfigDTO } from 'services/cd-ng'
import { ElkConnectorAuthFieldsProps } from './CreateElkConnector.types'
import { ELKConnectorFields, ElkAuthType } from './CreateElkConnector.constants'

type InitializeElkConnectorArgs = {
  prevStepData?: ConnectorConfigDTO
  accountId: string
  projectIdentifier?: string
  orgIdentifier?: string
}

export function initializeElkConnector({
  prevStepData,
  accountId,
  projectIdentifier,
  orgIdentifier
}: InitializeElkConnectorArgs): FormData {
  const defaultObj = {
    url: '',
    accountId,
    username: '',
    authType: AuthTypes.USER_PASSWORD,
    password: undefined,
    projectIdentifier,
    orgIdentifier
  }

  if (!prevStepData) {
    return defaultObj
  }
  const { spec, ...prevData } = prevStepData
  return {
    ...defaultObj,
    url: prevData?.url || spec?.controllerUrl || spec?.url || '',
    password: prevData?.password || spec?.passwordRef,
    clientSecretRef: prevData?.clientSecretRef || spec?.clientSecretRef,
    authType: prevData?.authType || spec?.authType || AuthTypes.USER_PASSWORD,
    username: prevData?.username || spec?.username || '',
    clientId: prevData?.clientId || spec?.clientId,
    apiKeyId: prevData?.apiKeyId || spec?.apiKeyId,
    apiKeyRef: prevData?.apiKeyRef || spec?.apiKeyRef
  }
}

interface GetElkAuthTypeReturn {
  label: string
  value: string
}

export const getELKAuthType = (getString: UseStringsReturn['getString']): GetElkAuthTypeReturn[] => [
  {
    label: getString('usernamePassword'),
    value: AuthTypes.USER_PASSWORD
  },
  {
    label: getString('common.apikey'),
    value: AuthTypes.API_CLIENT_TOKEN
  },
  {
    label: getString('platform.connectors.bearerToken'),
    value: AuthTypes.BEARER_TOKEN
  },
  {
    label: getString('platform.connectors.elk.noAuthentication'),
    value: ElkAuthType.NONE
  }
]

export const getAuthFields = (
  getString: UseStringsReturn['getString'],
  authType?: string
): ElkConnectorAuthFieldsProps[] | null => {
  if (!authType) {
    return null
  }

  switch (authType) {
    case AuthTypes.USER_PASSWORD:
      return [
        { name: ELKConnectorFields.USERNAME, label: getString('username'), key: ELKConnectorFields.USERNAME },
        { name: ELKConnectorFields.PASSWORD, label: getString('password'), key: ELKConnectorFields.PASSWORD }
      ]

    case AuthTypes.API_CLIENT_TOKEN:
      return [
        {
          name: ELKConnectorFields.API_KEY_ID,
          label: getString('platform.connectors.elk.apiId'),
          key: ELKConnectorFields.API_KEY_ID
        },
        {
          name: ELKConnectorFields.API_KEY_REF,
          label: getString('common.apikey'),
          key: ELKConnectorFields.API_KEY_REF
        }
      ]

    case AuthTypes.BEARER_TOKEN:
      return [
        {
          name: ELKConnectorFields.API_KEY_REF,
          label: getString('platform.connectors.bearerToken'),
          key: ELKConnectorFields.API_KEY_REF
        }
      ]

    default:
      return null
  }
}
