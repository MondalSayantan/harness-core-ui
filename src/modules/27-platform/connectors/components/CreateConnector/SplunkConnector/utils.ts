/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { FormData } from '@platform/connectors/interfaces/ConnectorInterface'
import { AuthTypes } from '@platform/connectors/pages/connectors/utils/ConnectorHelper'
import type { ConnectorConfigDTO } from 'services/cd-ng'

type InitializeSplunkConnectorArgs = {
  prevStepData?: ConnectorConfigDTO
  accountId: string
  projectIdentifier?: string
  orgIdentifier?: string
}

export function initializeSplunkConnector({
  prevStepData,
  projectIdentifier,
  accountId,
  orgIdentifier
}: InitializeSplunkConnectorArgs): FormData {
  const defaultObj = {
    url: '',
    authType: AuthTypes.USER_PASSWORD,
    username: '',
    passwordRef: undefined,
    tokenRef: undefined,
    accountId: accountId,
    projectIdentifier,
    orgIdentifier
  }

  if (!prevStepData) {
    return defaultObj
  }

  const { spec, ...prevData } = prevStepData
  return {
    ...defaultObj,
    url: prevData?.url || spec?.splunkUrl || '',
    username: prevData?.username || spec?.username,
    passwordRef: prevData?.passwordRef || spec?.passwordRef,
    authType: prevData?.authType || spec?.type || defaultObj.authType,
    tokenRef: prevData?.tokenRef || spec?.tokenRef
  }
}
