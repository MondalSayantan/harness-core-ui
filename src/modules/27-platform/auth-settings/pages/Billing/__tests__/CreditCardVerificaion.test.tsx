/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, RenderResult, waitFor } from '@testing-library/react'
import { loadStripe } from '@stripe/stripe-js'
import { Elements } from '@stripe/react-stripe-js'
import userEvent from '@testing-library/user-event'
import { TestWrapper } from '@common/utils/testUtils'
import { useGetDefaultCard, useCreateClientSecret, useSaveCard } from 'services/cd-ng'
import { Editions, TimeType } from '@common/constants/SubscriptionTypes'
import CreditCardVerification from '../CreditCardVerification'
jest.mock('services/cd-ng')

const useGetPrimaryCarddMock = useGetDefaultCard as jest.MockedFunction<any>
const useCreateClientSecretMock = useCreateClientSecret as jest.MockedFunction<any>
const useSaveCardMock = useSaveCard as jest.MockedFunction<any>
useCreateClientSecretMock.mockImplementation(() => {
  return {
    cancel: jest.fn(),
    loading: false,
    mutate: jest.fn().mockImplementationOnce(() => {
      return {
        status: 'SUCCESS',
        data: 'sdaads'
      }
    })
  }
})

useSaveCardMock.mockImplementation(() => {
  return {
    mutate: jest.fn()
  }
})
useGetPrimaryCarddMock.mockImplementation(() => {
  return {
    data: '',
    refetch: jest.fn()
  }
})
const clientSecret = 'dummy secret'
const stripePromise = loadStripe('dummy promise')

describe('billing page payment method', () => {
  const setSubscriptionProps = jest.fn()
  let renderObj: RenderResult
  beforeEach(() => {
    renderObj = render(
      <TestWrapper>
        <Elements stripe={stripePromise} options={{ clientSecret }}>
          <CreditCardVerification
            subscriptionProps={{
              edition: Editions.ENTERPRISE,
              premiumSupport: true,
              paymentFreq: TimeType.MONTHLY,
              subscriptionId: 'sasdasd',
              billingContactInfo: {
                name: 'sdas',
                email: 'dasaas',
                billingAddress: 'ddasdasd',
                city: 'dadsa',
                state: 'dadas',
                country: 'asds',
                zipCode: 'dsada',
                companyName: 'sdadas'
              },
              paymentMethodInfo: {
                paymentMethodId: 'dsdas',
                cardType: 'sdadas',
                expireDate: 'dsdasd',
                last4digits: 'dsdadad',
                nameOnCard: ''
              },
              productPrices: { monthly: [], yearly: [] },
              isValid: true
            }}
            onClose={jest.fn()}
            clientSecret={'asdas'}
            setView={jest.fn()}
            setInvoiceData={jest.fn()}
            setSubscriptionProps={setSubscriptionProps}
          />
        </Elements>
      </TestWrapper>
    )
  })
  afterEach(() => {
    renderObj.unmount()
  })
  test('render payment method card', async () => {
    const { container, getByTestId } = renderObj
    expect(container).toMatchSnapshot('credit card verification')

    const a = await waitFor(() => getByTestId('nameOnCard'))
    expect(a).toBeDefined()
    await waitFor(() => expect(getByTestId('nameOnCard')).toHaveValue(''))
    await userEvent.type(a, 'J')
    await waitFor(() => {
      expect(setSubscriptionProps).toHaveBeenCalledWith({
        billingContactInfo: {
          billingAddress: 'ddasdasd',
          city: 'dadsa',
          companyName: 'sdadas',
          country: 'asds',
          email: 'dasaas',
          name: 'sdas',
          state: 'dadas',
          zipCode: 'dsada'
        },
        edition: 'ENTERPRISE',
        isValid: true,
        paymentFreq: 'Monthly',
        paymentMethodInfo: {
          cardType: 'sdadas',
          expireDate: 'dsdasd',
          last4digits: 'dsdadad',
          nameOnCard: 'J',
          paymentMethodId: 'dsdas'
        },
        premiumSupport: true,
        productPrices: { monthly: [], yearly: [] },
        subscriptionId: 'sasdasd'
      })
    })
  })
})
describe('billing page payment method', () => {
  const setSubscriptionProps = jest.fn()
  let renderObj: RenderResult
  beforeEach(() => {
    renderObj = render(
      <TestWrapper>
        <Elements stripe={stripePromise} options={{ clientSecret }}>
          <CreditCardVerification
            subscriptionProps={{
              edition: Editions.ENTERPRISE,
              premiumSupport: true,
              paymentFreq: TimeType.MONTHLY,
              subscriptionId: 'sasdasd',
              billingContactInfo: {
                name: 'sdas',
                email: 'dasaas',
                billingAddress: 'ddasdasd',
                city: 'dadsa',
                state: 'dadas',
                country: 'asds',
                zipCode: 'dsada',
                companyName: 'sdadas'
              },
              paymentMethodInfo: {
                paymentMethodId: 'dsdas',
                cardType: 'sdadas',
                expireDate: 'dsdasd',
                last4digits: 'dsdadad',
                nameOnCard: ''
              },
              productPrices: { monthly: [], yearly: [] },
              isValid: true
            }}
            onClose={jest.fn()}
            clientSecret={'asdas'}
            setView={jest.fn()}
            setInvoiceData={jest.fn()}
            setSubscriptionProps={setSubscriptionProps}
          />
        </Elements>
      </TestWrapper>
    )
  })
  afterEach(() => {
    renderObj.unmount()
  })

  test('render payment method card with undefined paymentmethod info', async () => {
    const { container, getByTestId } = renderObj
    expect(container).toMatchSnapshot('credit card verification')

    const a = await waitFor(() => getByTestId('nameOnCard'))
    expect(a).toBeDefined()
    await waitFor(() => expect(getByTestId('nameOnCard')).toHaveValue(''))
    await userEvent.type(a, 'J')
    await waitFor(() => {
      expect(setSubscriptionProps).toHaveBeenCalledWith({
        billingContactInfo: {
          billingAddress: 'ddasdasd',
          city: 'dadsa',
          companyName: 'sdadas',
          country: 'asds',
          email: 'dasaas',
          name: 'sdas',
          state: 'dadas',
          zipCode: 'dsada'
        },
        edition: 'ENTERPRISE',
        isValid: true,
        paymentFreq: 'Monthly',
        paymentMethodInfo: {
          cardType: 'sdadas',
          expireDate: 'dsdasd',
          last4digits: 'dsdadad',
          nameOnCard: 'J',
          paymentMethodId: 'dsdas'
        },
        premiumSupport: true,
        productPrices: { monthly: [], yearly: [] },
        subscriptionId: 'sasdasd'
      })
    })
  })
})
