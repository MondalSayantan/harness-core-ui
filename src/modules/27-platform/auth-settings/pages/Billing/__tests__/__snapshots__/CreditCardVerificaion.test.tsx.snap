// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`billing page payment method render payment method card with undefined paymentmethod info: credit card verification 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main"
  >
    <div
      class="bp3-card bp3-elevation-0 Card--card paymentCard"
    >
      <div
        class="Layout--vertical StyledProps--main"
      >
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-start StyledProps--flex-justifyContent-space-between"
        >
          <div
            class="Layout--vertical StyledProps--main"
            style="width: 55%;"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--padding-bottom-large StyledProps--font-variation-h5"
            >
              platform.authSettings.billingInfo.paymentMethod
            </p>
          </div>
          <img
            alt=""
            aria-hidden="true"
            class="powerByStripe"
            src="test-file-stub"
          />
        </div>
        <div
          class="Layout--horizontal StyledProps--main paymentgrid"
        >
          <div
            class="Layout--vertical StyledProps--main"
            style="width: 35%;"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--padding-bottom-small"
            >
              common.nameOnCard
            </p>
            <div
              class="TextInput--main"
            >
              <div
                class="bp3-input-group"
              >
                <input
                  class="bp3-input"
                  data-testid="nameOnCard"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="Layout--vertical StyledProps--main"
            style="width: 55%;"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--padding-bottom-small"
            >
              common.cardNumber
            </p>
            <div
              class="creditCard"
            >
              <div />
            </div>
          </div>
        </div>
        <div
          class="Layout--horizontal StyledProps--main"
        >
          <label
            class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
          >
            <input
              data-testid="row-checkbox"
              type="checkbox"
            />
            <span
              class="bp3-control-indicator"
            />
          </label>
          <p
            class="StyledProps--font StyledProps--main warning StyledProps--font-size-small StyledProps--font-weight-bold StyledProps--padding-small"
          >
            platform.authSettings.billingInfo.saveCardWarning
          </p>
        </div>
      </div>
    </div>
    <div
      class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
    >
      <button
        class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--variation Button--variation-primary"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          platform.authSettings.setAsDefaultCard
        </span>
      </button>
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--variation Button--variation-secondary"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          cancel
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`billing page payment method render payment method card: credit card verification 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main"
  >
    <div
      class="bp3-card bp3-elevation-0 Card--card paymentCard"
    >
      <div
        class="Layout--vertical StyledProps--main"
      >
        <div
          class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-start StyledProps--flex-justifyContent-space-between"
        >
          <div
            class="Layout--vertical StyledProps--main"
            style="width: 55%;"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--padding-bottom-large StyledProps--font-variation-h5"
            >
              platform.authSettings.billingInfo.paymentMethod
            </p>
          </div>
          <img
            alt=""
            aria-hidden="true"
            class="powerByStripe"
            src="test-file-stub"
          />
        </div>
        <div
          class="Layout--horizontal StyledProps--main paymentgrid"
        >
          <div
            class="Layout--vertical StyledProps--main"
            style="width: 35%;"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--padding-bottom-small"
            >
              common.nameOnCard
            </p>
            <div
              class="TextInput--main"
            >
              <div
                class="bp3-input-group"
              >
                <input
                  class="bp3-input"
                  data-testid="nameOnCard"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="Layout--vertical StyledProps--main"
            style="width: 55%;"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--padding-bottom-small"
            >
              common.cardNumber
            </p>
            <div
              class="creditCard"
            >
              <div />
            </div>
          </div>
        </div>
        <div
          class="Layout--horizontal StyledProps--main"
        >
          <label
            class="bp3-control bp3-checkbox StyledProps--font Checkbox--checkbox StyledProps--main"
          >
            <input
              data-testid="row-checkbox"
              type="checkbox"
            />
            <span
              class="bp3-control-indicator"
            />
          </label>
          <p
            class="StyledProps--font StyledProps--main warning StyledProps--font-size-small StyledProps--font-weight-bold StyledProps--padding-small"
          >
            platform.authSettings.billingInfo.saveCardWarning
          </p>
        </div>
      </div>
    </div>
    <div
      class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
    >
      <button
        class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--variation Button--variation-primary"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          platform.authSettings.setAsDefaultCard
        </span>
      </button>
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--variation Button--variation-secondary"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          cancel
        </span>
      </button>
    </div>
  </div>
</div>
`;
